jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -429,6 +486,21 @@ public void setMaxParts(long maxParts)
             parser.setMaxParts(maxParts);
         }
 
+        /**
+         * Configure the Parser given a {@link MultiPartConfig} instance.
+         * @param config the configuration.
+         */
+        public void configure(MultiPartConfig config)
+        {
+            parser.setMaxParts(config.getMaxParts());
```

Comment:
So I'm wondering if we should add a method `TypeUtil.getClassHash(Class<?>)` that deterministically creates a hash of the signature of a class.  Then if it ever changes, we can know we need to update a method like this:
```
static
{
    if (TypeUtil.getClassHash(MultiPartConfig.class) != 0xF32B15790)
        throw new IllegalStateException("MultiPartConfig class has been updated");
 }
```

This could be used in several places in our code base.   Thoughts?

Reply:
I don't think this is necessary, this is the only place where we get the values from the `MultiPartConfig`.
So if the class is ever updated in the future should be simple to update the usage here.

Reply:
@lachlan-roberts the whole point of my proposal is that it is very easy to add fields to a class like `MultiPartConfig`, but forget to update a method like `configure(MultiPartConfig config)`.   My proposal would make a signature of a class and would fail if ever the class was modified.    It would be generally useful in many places where we manually copy a class field by field..... but orthogonal to this PR.

Reply:
I understand that, I just think in this case it would be unlikely to add a field/setter to `MultiPartConfig` but not update the single usage of `MultiPartConfig` which configures the parser, and have it get past any review, and also pass any tests added for it.

But if we start using a pattern like that more generally it should be applied here as well. 
Anyway something to consider in a separate PR.